standard/interfaces.md

```diff
@@ -242,24 +255,133 @@ If a `new` modifier is included in a declaration that doesn’t hide an inherite
 
+Consider an interface with a default implementation for a member `M`. As `M` is not part of that interface’s contract, outside that interface or any interface derived from it, that name is not visible. How then can it be accessed? The following code shows how:
+
+<!-- Example: {template:"standalone-console", name:"InterfaceMember", expectedOutput:["IB.M", "IA.P = 10", "IB.P = 20"]} -->
+```csharp
+interface IA
+{
+    public int P { get { return 10; } }
+    public void M()
+    {
+        Console.WriteLine("IA.M");
+    }
+}
+
+interface IB: IA
+{
+    public new int P { get { return 20; } }
+    void IA.M()
+    {
+        Console.WriteLine("IB.M");
+    }
+}
+
+class C: IB { }
+
+class Test
+{
+    public static void Main()
+    {
+        C c = new C();
+        ((IA)c).M();                               // cast needed
+        Console.WriteLine($"IA.P = {((IA)c).P}");  // cast needed
+        Console.WriteLine($"IB.P = {((IB)c).P}");  // cast needed
+    }
+}
+```
+
+Within the interfaces `IA` and `IB`, member `M` is accessible directly by name. However, within method `Start`, we cannot write `c.M()` or `c.P`, as those names are not visible. To find them, casts to the appropriate interface type are needed.
+
+> *Note*: See how the declaration of `M` in `IB` uses explicit interface implementation syntax. This is necessary to make that method override the one in `IA`; the modifier `override` may not be applied to a function member. *end note*
+
+### §interface-constants Interface constants
+
+[§15.4](classes.md#154-constants) covers constant declarations in classes. When declaring constants in interfaces, that text needs to be interpreted in the context of an interface, and, where necessary, it is augmented and/or replaced by text in this subclause.
+
+*constant_declaration* is described in [§15.4](classes.md#154-constants).
+
+As a *constant_declaration* is considered to have a default implementation ([§18.1](interfaces.md#181-general)), it is *not* part of the interface’s contract.
+
+See §interface-static-constructors for information regarding the allocation and initialization of constants.
+
+> *Note*: See §interface-fields for an example of using various kinds of static members declared within an interface. *end note*
+
+### §interface-fields Interface fields
+
+[§15.5](classes.md#155-fields) and its subclauses cover field declarations in classes. When declaring fields in interfaces, that text needs to be interpreted in the context of an interface, and, where necessary, it is augmented and/or replaced by text in this subclause.
+
+*field_declaration* is described in [§15.5.1](classes.md#1551-general).
+
+It is a compile-time error for *field_declaration* to declare an instance field.
+
+As a static *field_declaration* is considered to have a default implementation ([§18.1](interfaces.md#181-general)), it is *not* part of the interface’s contract.
```

Comment:
Here and elsewhere: do we want to talk about fields and constants as "having implementations"? Do we even want to talk about static members as having ***default*** implementations, as though there was any "slot" being "implemented" where we selected the "default" option for the slot?